### PR TITLE
Restructure user docs on MarkBind syntax

### PR DESCRIPTION
**Pull request by damithc**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**
• [x] Documentation update

This is another update to user docs, focusing on how we describe MarkBind syntax. Overview:

* Minor refinements to the landing page
* Extracted explanation of each syntax into a separate file in the `userGuide/syntax/` folder.
* Other pages that changed significantly:
  * `Authoring Contents`: added overviews of new pages
  * `MarkBind Syntax Overview`: Previously this page contained all syntax (except components). Not it contains an overview only.
  * `Formatting Contents`: new page. Describes all syntax used for basic formatting.
  * `Using Components`: Revised to pull content from syntax/*.mbdf pages
  * `Using HTML, JavaScript, CSS`: new page.
  * `Tweaking the Page Structure`: Revised to pull content from syntax/*.mbdf pages
  * `Reusing Contents`: Revised to pull content from syntax/*.mbdf pages
  * `Making the Site Searchable`: Revised to pull content from syntax/*.mbdf pages
  * References
    * `Reader-Facing Features`: New page. Shows features that will be visible to readers.
    * `Full Syntax Reference`: Alphabetical list of all syntax
    * `Syntax Cheat Sheet`: new page

### CODE REVIEW

**Comment by damithc** [comment]
Ready for review. As we did with the previous one, we can fix only major issues before merging. Minor issues can be fixed as followup PRs. It will take a few more iterations before the user docs are as polished as we want them to be.

**Comment by yamgent** [comment]
@Chng-Zhi-Xuan will need your help in reviewing this PR.

**Comment by damithc** [comment]
For some reason, the Netlify preview appears broken 🤔

**Comment by damithc** [comment]
Resolved merge conflict.

**Comment by damithc** [comment]
Looks like netlify is getting some erros:
```
1:37:40 PM:  v1.17.2
1:37:40 PM: info: Website generation started at 05:37:40
1:37:40 PM: info: Building assets...
1:37:40 PM: info: Assets built
1:37:40 PM: info: Generating pages...
1:37:40 PM: error: Error: No such file: /opt/build/repo/docs/userGuide/markbindSyntaxOverview.md
1:37:40 PM: Missing reference in /opt/build/repo/docs/userGuide/authoringContents.md
1:37:40 PM: error: Error: No such file: /opt/build/repo/docs/userGuide/markbindSyntaxOverview.md
1:37:40 PM: Missing reference in /opt/build/repo/docs/userGuide/authoringContents.md
1:37:40 PM: error: Template render error: (unknown path)
1:37:40 PM:   Template render error: (unknown path)
1:37:40 PM:   Error: template not found: userGuide/fullSyntaxReference.md
1:37:40 PM: Error while preprocessing '/opt/build/repo/docs/userGuide/authoringContents.md'
1:37:40 PM: error: Error: No such file: /opt/build/repo/docs/userGuide/syntax/navBars.mbdf
1:37:40 PM: Missing reference in /opt/build/repo/docs/userGuide/fullSyntaxReference.md
1:37:40 PM: error: Error: No such file: /opt/build/repo/docs/userGuide/syntax/searchBars.mbdf
1:37:40 PM: Missing reference in /opt/build/repo/docs/userGuide/fullSyntaxReference.md
1:37:40 PM: error: Template render error: (unknown path)
1:37:40 PM:   Template render error: (unknown path)
1:37:40 PM:   Error: template not found: userGuide/fullSyntaxReference.md
1:37:40 PM: Error while preprocessing '/opt/build/repo/docs/userGuide/markBindSyntaxOverview.md'
```

File: docs/userGuide/authoringContents.md [hunk 1]
```diff
@@ -10,67 +10,28 @@
...
-
-<include src="reusingContents.md#overview" inline />
-</blockquote>
-
-&nbsp;&nbsp;&nbsp;&nbsp; [Read more ...]({{ baseUrl }}/userGuide/reusingContents.html)
-
 
+{% set pages = [
+  ['Adding Pagess', 'addingPages'],
+  ['MarkBind Syntax Overview', 'markbindSyntaxOverview'],
```
Comment (nicholaschuayunzhi) [inline docs/userGuide/authoringContents.md]:
```
  ['Adding Pages', 'addingPages'],
  ['MarkBind Syntax Overview', 'markbindSyntaxOverview'],
```

**Comment by damithc** [comment]
Let's merge this if there are no big problems -- if not, it will create merge conflict headaches for all the incoming PRs. No need to review content itself as I will be submitting more PRs to refine content further. We can do a fine-grain review of the text when the content is stable.

**Comment by yamgent** [comment]
On a side note for performance: Build time for documentation seems to have doubled since this PR was merged (was around ~3.174s before, now it is ~7.697s).

**Comment by damithc** [comment]
> On a side note for performance: Build time for documentation seems to have doubled since this PR was merged (was around ~3.174s before, now it is ~7.697s).

Due to more source files, and more output files? 🤔 There are some nunjucks scripts too.

**Pull request merged**